### PR TITLE
Enable multiple results to be returned from <google-map-search>

### LINKED ISSUE #16: google-map-search only returns 1 pseudo result
google-map-search only returns 1 result, and it only contains the latlng and a boolean, users can't access the place name or any of the other exciting [PlaceResult](https://developers.google.com/maps/documentation/javascript/reference#PlaceResult) fields they might want to use.

Alternatively, google-map-search could stay as is to support current users and a google-map-places element could be built that's a bit more robust.

### PR DESCRIPTION
**Pull request by PandaWhisperer**
## Summary

This allows multiple results to be returned from `<google-map-search>` (instead of just one). This fixes #16.

## Details

This is the simplest way I could come up with to generalize the `<google-map-search>` tag for multiple results. 

The `result` property and accompanying event have been renamed to `results`, and `google-map-search-results>`, respectively. All results are processed so that they have an obvious `latitude` and `longitude` property (instead of having to obtain that information manually via `geometry`). No other post-processing is done.

The documentation, including the usage example, has been updated accordingly. The example now shows how to display the search result on the actual map using `<google-map-marker>` elements. 

## Features

- Restrict the search to the currently visible map area using the `currentMapArea` attribute
- Restrict search results to show only certain result types via the `types` attribute
- Search by location and radius (location defaults to current map center)
- Documentation and updated example included

## Known Issues

~~Even with the `map` property set, the search is currently not affected by the location shown on the map, or restricted to the map boundaries. Instead, it shows the top results world wide (I believe). I plan to address this in the next version.~~

~~There is no support for other optional settings of the `PlacesService#textSearch` function, but I'm planning to add this soon.~~

The `currentMapArea` attribute does not work if the `map` is not yet fully loaded.

### CODE REVIEW

**Comment by PandaWhisperer** [comment]
CLA signed.

**Comment by PandaWhisperer** [comment]
Any feedback on this? This request has been open for a month now...

Thanks!

**Comment by ebidel** [comment]
This will need to be updated to the 1.0 element if you'd still like to see it.

**Comment by PandaWhisperer** [comment]
Rebased onto current master branch.

File: google-map-search.html [hunk 1]
```diff
@@ -57,36 +63,128 @@
...
+
+      /**
+       * Search radius, in meters.
+       * If `latitude` and `longitude` are not specified,
+       * the center of the currently visible map area is used.
+       *
+       * Ignored if `currentMapArea` is true.
+       *
+       * @attribute radius
+       * @type number
```
Comment (ebidel) [inline google-map-search.html]: you can remoe `@type` from the doc strings

File: google-map-search.html [hunk 2]
```diff
@@ -57,36 +63,128 @@
...
+      },
+
+      /**
+       * Search radius, in meters.
+       * If `latitude` and `longitude` are not specified,
+       * the center of the currently visible map area is used.
+       *
+       * Ignored if `currentMapArea` is true.
+       *
+       * @attribute radius
```
Comment (ebidel) [inline google-map-search.html]: remove @ attribute from the doc strings

File: google-map-search.html [hunk 3]
```diff
@@ -57,36 +63,128 @@
...
+        type: Number,
+        value: null
+      },
+
+      /**
+       * Restrict the search only to the current map area.
+       *
+       * @attribute currentMapArea
+       */
+      currentMapArea: {
```
Comment (ebidel) [inline google-map-search.html]: how about `searchCurrentMapArea`

File: google-map-search.html [hunk 4]
```diff
@@ -57,36 +63,128 @@
...
+       */
+      radius: {
+        type: Number,
+        value: null
+      },
+
+      /**
+       * Restrict the search only to the current map area.
+       *
+       * @attribute currentMapArea
```
Comment (ebidel) [inline google-map-search.html]: remove

File: google-map-search.html [hunk 5]
```diff
@@ -57,36 +63,128 @@
...
+       */
+      currentMapArea: {
+        type: Boolean,
+        value: true
```
Comment (ebidel) [inline google-map-search.html]: for booleans, it's better to have the default value be false. When the attribute is applied, it becomes thruthy.

File: google-map-search.html [hunk 6]
```diff
@@ -57,36 +63,128 @@
...
+        type: String,
+        value: null
+      },
+
+      /**
+       * The search results.
+       */
+      results: {
+        type: Array,
         value: null,
```
Comment (ebidel) [inline google-map-search.html]: `value: function() { return []; }`

File: google-map-search.html [hunk 7]
```diff
@@ -57,36 +63,128 @@
...
-      'search(query,map)'
+      'search(query,map,location,radius,types,currentMapArea)',
+      '_updateLocation(latitude,longitude)'
     ],
 
     /**
-     * Performance a search using for `query` for the search term.
+     * Perform a search using for `query` for the search term.
+     *
+     * @method search
```
Comment (ebidel) [inline google-map-search.html]: remove

File: google-map-search.html [hunk 8]
```diff
@@ -57,36 +63,128 @@
...
     search: function() {
       if (this.query && this.map) {
         var places = new google.maps.places.PlacesService(this.map);
-        places.textSearch({query: this.query}, this._gotResults.bind(this));
+
+        if (this.types && typeof this.types == 'string') {
+          var types = this.types.split(' ');
+        }
+
+        if (this.currentMapArea) {
```
Comment (ebidel) [inline google-map-search.html]: you'll need to update these checks if you swap the boolean to default to false

File: google-map-search.html [hunk 9]
```diff
@@ -57,36 +63,128 @@
...
     _gotResults: function(results, status) {
-      this.result = {
-        latitude: results[0].geometry.location.lat(),
-        longitude: results[0].geometry.location.lng(),
-        show: true
+      this.results = results.map(function(result) {
+        // obtain lat/long from geometry
+        result.latitude  = result.geometry.location.lat();
+        result.longitude = result.geometry.location.lng();
+
```
Comment (ebidel) [inline google-map-search.html]: not sure why we had `show: true`, but why did you remove it?

**Comment by ebidel** [comment]
Reviewed. Sorry for the delay

**Comment by PandaWhisperer** [comment]
Thanks for the feedback, @ebidel. 

I'm okay with all the proposed changes, except renaming `currentMapArea` to `dontSearchCurrentMapArea`, because negations in boolean variable names can be confusing. I named it `global` instead. (I'm open to naming it `searchGlobal` or `globalSearch` if you feel that's a better choice).

**Comment by ebidel** [comment]
`globalSearch` seems good and descriptive. Can you update and we'll merge!

**Comment by PandaWhisperer** [comment]
Updated.

**Comment by ebidel** [comment]
Thanks
